rrs/views.py

```diff
@@ -0,0 +1,1005 @@
+
+    writer = csv.writer(response)
+    writer.writerow(['Upstream status', 'Name', 'Version',
+        'Upstream version', 'Maintainer', 'Summary'])
+    for r in recipe_list:
+        writer.writerow([r.upstream_status, r.name, r.version,
+            r.upstream_version, r.maintainer_name.encode('utf-8'), r.summary])
+
+    return response
+
+
+
+class RecipeUpgradeGroupSortItem:
+    def __init__(self, group):
+        self.group = group
+        self.ver = RecipeUpgradeGroupSortItem.group_to_ver(self.group)
+
+    @staticmethod
+    def group_to_ver(grp):
+        if not grp:
+            return []
+        else:
+            try:
+                return list(map(int, grp.title.split('.')))
+            except ValueError:
+                return grp.recipeupgrade_set.exclude(upgrade_type__in=['N','R']).order_by('-id').values_list('id', flat=True).first()
+
+    def __lt__(self, other):
+        if type(self.ver) == type(other.ver):
+            return self.ver < other.ver
+        elif isinstance(self.ver, str) and isinstance(other.ver, list):
+            return True
+        elif isinstance(self.ver, int) and isinstance(other.ver, int):
+            return False
+
+    def __repr__(self):
+        return repr(self.group)
+
+
+
+class RecipeUpgradeDetail():
+    title = None
+    version = None
+    maintplan_name = None
+    release_name = None
+    milestone_name = None
+    date = None
+    maintainer_name = None
+    is_recipe_maintainer = None
+    commit = None
+    commit_url = None
+    upgrade_type = None
+    group = None
+
+    def __init__(self, title, version, maintplan_name, release_name, milestone_name, date,
+            maintainer_name, is_recipe_maintainer, commit, commit_url, upgrade_type,
+            group):
+        self.title = title
+        self.version = version
+        self.maintplan_name = maintplan_name
+        self.release_name = release_name
+        self.milestone_name = milestone_name
+        self.date = date
+        self.maintainer_name = maintainer_name
+        self.is_recipe_maintainer = is_recipe_maintainer
+        self.commit = commit
+        self.commit_url = commit_url
+        self.upgrade_type = upgrade_type
+        self.group = group
+
+def _get_recipe_upgrade_detail(maintplan, recipe_upgrade):
+    release_name = ''
+    milestone_name = ''
+    recipe_maintainer_history = None
+
+    release = Release.get_by_date(maintplan, recipe_upgrade.commit_date)
+    if release:
+        release_name = release.name
+        milestone = Milestone.get_by_release_and_date(release,
+                recipe_upgrade.commit_date)
+        if milestone:
+            milestone_name = milestone.name
+            recipe_maintainer_history = RecipeMaintainerHistory.get_by_end_date(
+                recipe_upgrade.recipesymbol.layerbranch,
+                milestone.end_date)
+
+    is_recipe_maintainer = False
+    maintainer_name = ''
+    if not recipe_upgrade.maintainer is None:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Use `is not` operator rather than `not ... is` because it is more readable and makes your code easier to understand.

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=Use%20is%20not)

Similar issue at line number 704.